src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.ExtensionMethodSymbolComputer.cs

```diff
@@ -65,37 +73,36 @@ private class ExtensionMethodSymbolComputer
                     document, semanticModel, receiverTypeSymbol, position, namespaceInScope);
             }
 
+            private static string? GetPEReferenceCacheKey(PortableExecutableReference peReference)
+                => peReference.FilePath ?? peReference.Display;
+
             /// <summary>
-            /// Force create all relevant indices
+            /// Force create/update all relevant indices
             /// </summary>
-            public static Task PopulateIndicesAsync(Project? project, IImportCompletionCacheService<CacheEntry, object> cacheService, CancellationToken cancellationToken)
+            public static async Task UpdateCacheAsync(Project? project, CancellationToken cancellationToken)
             {
-                if (project is null)
+                if (project is not null)
                 {
-                    return Task.CompletedTask;
-                }
+                    var cacheService = GetCacheService(project.Solution.Workspace);
 
-                using var _ = ArrayBuilder<Task>.GetInstance(out var tasks);
+                    foreach (var relevantProject in GetAllRelevantProjects(project))
+                        await GetUpToDateCacheEntryAsync(relevantProject, cacheService, cancellationToken).ConfigureAwait(false);
 
-                foreach (var relevantProject in GetAllRelevantProjects(project))
-                {
-                    tasks.Add(Task.Run(()
-                        => GetCacheEntryAsync(relevantProject, loadOnly: false, cacheService, cancellationToken), cancellationToken));
+                    foreach (var peReference in GetAllRelevantPeReferences(project))
+                        await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(project.Solution, peReference, loadOnly: false, cancellationToken).ConfigureAwait(false);
                 }
+            }
```

Comment:
does thsi run in oop?  or in VS?  If in OOP, i'm ok with this running in parallel over all the PERefs.  Unless you want it serial? (i can see args both ways).

Reply:
This is in OOP (if enabled, that is) We can switch to serial too, now that we are more relaxed on returning outdated data or nothing if cache is empty, the actual work happening in foreground is minimal so parallelization might not provide much of a benefit anymore.